python-threatexchange/threatexchange/cli/hash_cmd.py

```diff
@@ -118,7 +144,17 @@ def execute(self, settings: CLISettings) -> None:
         if not self.rotations:
             for file in self.files:
                 for hasher in hashers:
-                    hash_str = hasher.hash_from_file(file)
+                    if isinstance(hasher, PdqSignal) and (
```

Comment:
Why only PdqSignal? Wouldn't other image perceptual hashing algorithms benefit from this?

We also generally want to avoid places where we do `isinstance(<interface_implementation>)` in preference of it being handled by the interface itself.

Reply:
Okay, I thought it was specific to PdqSignal and as for why I bypassed the interface it is because they do not all have the method hash from bytes and I did not want always to create the new file with updated images bytes. Is there a way I can go around this or would it be better to always create the new file even if temporarily and then save the output if the user passes the flag to save it?

Reply:
It's not specific to PdqSignal, and the hash_from_bytes method is itself part of a wider interface.

I think I eventually decided it was simpler to write everything to tmpfiles, which is how we ended up with the current implementation.

However, similar to the feedback I gave for --rotations, we can make our life a lot easier by having the preprocessing happen in between the file input and the `for file in self.files`.

I like your idea of providing a way to pass flag to save it, but let's save that for a followup.